### ISSUE TITLE
LoRA Training Stuck with No Errors (Google Colab, CUDA 12.2)

### ISSUE THREAD

**Issue by Atsushi888** [bmaltais/kohya_ss#3041]
Problem Description

While attempting to train a LoRA model using the Kohya-ss GUI, the training process does not progress or complete. No error messages are displayed, and the process appears to run indefinitely without any results. I would like to determine whether this issue is due to a configuration mistake or a potential bug in the application.

Environment Information
	•	Platform: Google Colab Pro
	•	Python Version: 3.10
	•	CUDA Version: 12.2
	•	GPU: NVIDIA L4
	•	Kohya-ss Version: Latest from GitHub (e5e8be05fe0475a04e61ef668afffc632aa178f5)
	•	Primary Libraries:
	•	torch==2.1.2
	•	tensorflow==2.17.1
	•	diffusers[torch]==0.25.1
	•	huggingface-hub==0.25.1
	•	safetensors==0.4.2

Steps Taken
	1.	Configured the necessary parameters in the Kohya-ss GUI as shown in the attached screenshot.
	•	Resolution (--resolution): 1024x1024
	2.	Started training from the GUI.
	3.	Verified CUDA version using nvcc --version, confirming it is version 12.2.
	4.	Observed that the problem consistently occurs under the same conditions.

Additional Information
	•	The attached screenshot illustrates the GUI state during the issue, including the parameters entered and the status when the process hangs.
	•	This issue arose as part of an effort to create a custom launcher script for Kohya-ss, designed to make it easier for users to operate the GUI in Google Colab.
	•	The goal is to simplify usage for beginners and non-technical users, ultimately increasing the accessibility and utility of Kohya-ss.
	•	If necessary, I am willing to share the source code of the script being developed to assist in reproducing and diagnosing the issue.
	•	The runtime environment in Google Colab is always configured with NVIDIA L4 GPUs.

Questions
	1.	Is this issue likely caused by a configuration error, or does it stem from a bug in the script or environment?
	2.	What are the recommended configurations or troubleshooting steps to resolve this issue?
	3.	If this is a known problem, is there a fix or a planned update to address it?

![IMG_4656](https://github.com/user-attachments/assets/1fc2e6d6-9c44-4f34-a101-94b0b5244d30)

**Comment by Sylsatra** [comment]
Umm, i think the problem coming from the fact that you trained a Lora with 1 epoch of 50 steps while you set the max train steps to 1000 and learning rate warmup steps to 100. 
Maybe it is supposed to be 10 epochs?

**Comment by Atsushi888** [comment]
![IMG_4661](https://github.com/user-attachments/assets/7dd94770-89f5-4941-b136-c85443f634c7)
![IMG_4662](https://github.com/user-attachments/assets/d3cce979-96f2-4e20-b36f-d431089a2d4a)
Apologies for the interruption earlier. Let me send the revised message again.

Thank you very much for your prompt response.
First, please allow me to share a screenshot of the parameters I am using to run the script.
(I am training 5 images, each repeated 10 times.)
These are the only parameters I have modified or interacted with.

I have adjusted the parameters and tested the setup multiple times, but unfortunately, the situation remains unchanged. This has been rather discouraging.
At this point, I suspect there might be an issue or error in the Kohya_ss GUI itself.

If you have any advice or suggestions to help me overcome this situation, I would greatly appreciate your support. Thank you again for your time and assistance.

**Comment by Sylsatra** [comment]
I think you should click the preset tab and choose an SDXL preset, then change the image, model, and output directory and you will be good to go. Feel free to ask me!

![Screenshot 2025-01-07 185538](https://github.com/user-attachments/assets/27cfd672-2a6d-4992-9125-d4f560ff9953)

**Comment by Atsushi888** [comment]
Hello,
Thank you very much for your kind advice. I deeply appreciate it.

However, I regret to inform you that changing the Presets as instructed did not result in any change in the outcome.
I also tried several other SDXL-related options available in the Presets, but unfortunately, the situation remained the same.

I have attached a screenshot of one of the Presets I tested. Please take a look and let me know if there is anything I might have overlooked.

I would greatly appreciate it if you could provide further advice to help me resolve this issue.

Thank you for your time and assistance.
![IMG_4667](https://github.com/user-attachments/assets/9ed03fc6-0673-4b5e-be84-cf5dc8098075)
![IMG_4665](https://github.com/user-attachments/assets/d69a6d2e-84b8-4b15-8e71-5985eb80084f)
![IMG_4666](https://github.com/user-attachments/assets/8289e51a-3b27-48ca-8b34-1f844dd8788a)

**Comment by Sylsatra** [comment]
At this point, the issue lies in the version of CUDA. I suggest you reinstall it from scratch using the master branch. The CUDA version should be 11.8! If you use another branch the requirements will differ.

**Comment by Atsushi888** [comment]
![IMG_4670](https://github.com/user-attachments/assets/68af36d0-3ad4-482c-91f8-72d2f19bdec0)

Dear Sylsatra,

I hope this message finds you well. I am writing to seek urgent assistance with an issue I am encountering while using the Kohya_ss GUI on Google Colab for LoRA training. Despite following all recommended steps and spending considerable time troubleshooting, I have been unable to achieve my intended goal.
	1.	Extensive Efforts and Technical Adaptations
I have diligently worked on implementing numerous technical solutions, including switching to CUDA 11.8. This involved resolving complex errors, configuring paths, and making several system-level changes. These tasks required significant technical knowledge, but I put in the effort to meet the requirements outlined for using Kohya_ss on Google Colab.
	2.	Time-Consuming Process
The process of understanding, implementing, and testing these changes has consumed an enormous amount of time. My dedication to resolving these issues reflects my commitment to using Kohya_ss for LoRA generation.
	3.	Deep Disappointment Despite Genuine Efforts
Despite my best efforts and successfully passing all checks (as shown in the attached screenshot), I am unable to accomplish LoRA training. This has caused me considerable frustration and disappointment, as I feel that I have done everything within my power to meet the necessary requirements.

Given the amount of time, effort, and resources I have dedicated to resolving these issues, I kindly request a prompt and comprehensive response from your team to help me achieve LoRA training on Google Colab. Your guidance on overcoming the challenges I am facing will be greatly appreciated.

I would also like to emphasize that, as a dedicated user of Kohya_ss, I am reaching out to you in the hope of receiving a professional and timely resolution to this issue.

Thank you for taking the time to review my case. I look forward to your response.

Sincerely,
Atsushi888

**Comment by Sylsatra** [comment]
I'll try using google colab later today and try to find out what is the problem

**Comment by Atsushi888** [comment]
Thank you very much. I truly appreciate your assistance. Custom LoRA will play a key role in my future creative endeavors.

**Comment by b-fission** [comment]
Put this snippet in your notebook before starting the GUI or training commands.

`os.environ["TERM"]="dumb"`

**Comment by b-fission** [comment]
For reference, here is the launcher code I made while exploring Colab. You might find it useful.
It automatically installs the necessary components and starts the GUI.

<details>
<summary>Launcher script for Colab (click this)</summary>

```python
import os

BASEPATH = "/content"

# important environment vars
GUIPATH = os.path.join(BASEPATH, "kohya_ss")
os.environ["TERM"] = "dumb"
os.environ["PATH"] = os.path.join(GUIPATH, "venv/bin") + ":" + os.environ["PATH"]
os.environ["LD_LIBRARY_PATH"] += ":{0}/torch/lib:{0}/nvidia/cusparse/lib:{0}/nvidia/cuda_runtime/lib".format(os.path.join(GUIPATH, "venv/lib/python3.10/site-packages"))

# installation
if not os.path.isdir(GUIPATH):
  # repo setup
  os.chdir(BASEPATH)
  !git clone https://github.com/bmaltais/kohya_ss.git
  os.chdir(GUIPATH)
  !git submodule update --init --recursive

  # prepare venv and install requirements
  !apt install python3.10-venv
  !python -m venv venv
  !./venv/bin/python -m pip install -U pip
  !./venv/bin/pip install nvidia-cusparse-cu11 nvidia-cuda_runtime-cu11
  !./setup.sh -v

# start the gui
os.chdir(GUIPATH)
!./gui.sh --headless --share

```
</details>


<details>
<summary>And a test showing the log output for a short training run.</summary>

![colab-kohya1](https://github.com/user-attachments/assets/2dc1f1f9-792c-48e3-aed3-1b4839904263)

</details>

**Comment by Atsushi888** [comment]
Dear Kohya Community,

I am writing to express my deepest gratitude for your invaluable guidance and support. After carefully reviewing my environment, I discovered that both cuDNN and CuPy were not functioning as expected. By dedicating time and effort to address these issues and incorporating the advice provided in your responses, I was finally able to achieve success in creating my original LoRA. This achievement has brought me immense satisfaction and fulfillment, and it would not have been possible without your insights and encouragement.

Furthermore, the idea of running Kohya in a virtualized environment was entirely new to me, and I can already foresee the significant advantages it offers. I am truly grateful for sharing this innovative approach with me, as it opens new doors for exploration and improvement.

Once again, thank you for your generous assistance and for sharing your expertise. I look forward to continuing my work with Kohya and contributing to this amazing community in the future.

Best regards,
Atsushi888

**Comment by dzd11111** [comment]
I have the same issues but im using runpod
